cpp/src/replace/clamp.cu

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/types.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/detail/copy.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/replace.hpp>
+#include <cudf/detail/iterator.cuh>
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+namespace {
+struct dispatch_clamp {
+    template <typename T, typename Predicate>
+    void apply_transform (column_device_view  input,
+                          mutable_column_device_view output,
+                          Predicate pred,
```

Comment:
`Predicate` is a bit of a misnomer here. Typically predicates are functions that return `true/false` used to filter an input. Consider something like `Transformer` instead, or just `F`.